tests/test_task.py

```diff
@@ -157,15 +157,39 @@ def test_build_trials_seq_maximum_sequential_trials():
     task._rng = np.random.default_rng(NTRIALS)
     modality_seq = task._build_trials_seq()
     # Ensure that no more than the specified maximum number of consecutive trials of the same modality occur
-    for modality in task.modalities:
+    for modality in task._modalities:
         for i in range(len(modality_seq) - task.max_sequential):
             assert np.sum(modality_seq[i : i + task.max_sequential] == modality) <= task.max_sequential
 
 
 def test_generate_trials(task: Task):
     trials = task.generate_trials()
-    assert trials["inputs"].shape == (task._ntrials, len(task.time), task.n_inputs)
-    assert trials["outputs"].shape == (task._ntrials, len(task.time), task.n_outputs)
+    assert trials["inputs"].shape == (task._ntrials, len(task._time), task._n_inputs)
+    assert trials["outputs"].shape == (task._ntrials, len(task._time), task.n_outputs)
+
+
+@pytest.mark.parametrize(
+    ("session", "ntrials", "random_seed"),
+    [
+        ({"v": 0.5, "a": 0.5}, 20, None),
+        ({"v": 1, "a": 3}, NTRIALS, 100),
+    ],
+)
+def test_reproduce_experiment(
+    session: dict,
+    ntrials: int,
+    random_seed: int | None,
+):
+    task = Task(name=NAME, session=session)
+    trials = task.generate_trials(ntrials=ntrials, random_seed=random_seed)
+    task_repro = Task(**trials["task_settings"])
+    trials_repro = task_repro.generate_trials(trials["ntrials"], trials["random_seed"])
+    # Check that the output is the same
+    assert np.array_equal(trials["modality_seq"], trials_repro["modality_seq"])
+    assert np.array_equal(trials["time"], trials_repro["time"])
+    assert all(np.array_equal(trials["phases"][key], trials_repro["phases"][key]) for key in trials["phases"])
+    assert np.array_equal(trials["inputs"], trials_repro["inputs"])
+    assert np.array_equal(trials["outputs"], trials_repro["outputs"])
```

Comment:
How about this? Automatically safeguards for additional settings that may be added to `Task` later on.

```suggestion

    # Check that the output is the same
    assert task == task_repro
    for x, y in trials.items():
        if x != "phases":  # tested separately because it's a dict of numpy arrays
            assert np.array_equal(y, trials_repro[x])
    assert all(np.array_equal(trials["phases"][key], trials_repro["phases"][key]) for key in trials["phases"])
```